packages/grpc-js/src/server.ts

```diff
@@ -1421,41 +1420,72 @@ export class Server {
       }
 
       if (this.keepaliveTimeMs < KEEPALIVE_MAX_TIME_MS) {
-        keeapliveTimeTimer = setInterval(() => {
-          keepaliveTimeoutTimer = setTimeout(() => {
-            sessionClosedByServer = true;
-            session.close();
+        keepaliveInterval = setInterval(() => {
+          const keepaliveTimeout = setTimeout(() => {
```

Comment:
Instead of making these changes to support multiple outstanding pings and corresponding keepalive timeout timers, I think it would be better to take the same approach as on the client, and ensure that there is only one outstanding ping at any given time.

There isn't really a reason to have multiple outstanding pings: in terms of keeping the connection alive, if you need to send a second ping before receiving a previous response, that implies that your round trip latency can be longer than the maximum connection age, which probably means that you have other problems. And in terms of detecting that a connection is no longer working, the timeout for the first ping will end before the timeout for the second ping, so the second ping doesn't let you find out faster.